tests/test_core_check.py

```diff
@@ -111,6 +113,20 @@ def test_check_run_save(tmp_path, retry_count):
     # Ensure content looks okay
     for line in lines[1:]:
         url, result, filename = line.split(",")
+
+        root = filename.split('/')[0]
         assert url.startswith("http")
         assert result in ["passed", "failed"]
         assert re.search("(.py|.md)$", filename)
```

Comment:
I noticed that you are using a lot of `assert ...``which will cause some warnings from Codacy, mentioning that it is better to use: ```if not ...: raise AssertionError `. We can keep it the way it is for now, but -I think- we will probably need to change this in the future.

Reply:
Ok! Sorry about that. We can definitely change it in the future, I’ve mostly been avoiding warnings from pytest and these seemed ok.